src/expression.d

```diff
@@ -11821,6 +11842,26 @@ extern (C++) final class CommaExp : BinExp
     {
         v.visit(this);
     }
+
+    /**
+     * If the argument is a CommaExp, set a flag to prevent deprecation messages
+     *
+     * It's impossible to know from CommaExp.semantic if the result will
+     * be used, hence when there is a result (type != void), a deprecation
+     * message is always emitted.
+     * However, some construct can produce a result but won't use it
+     * (ExpStatement and for loop increment).  Those should call this function
+     * to prevent unwanted deprecations to be emitted.
+     *
+     * Params:
+     *   exp = An expression that discards its result.
+     *         If the argument is null or not a CommaExp, nothing happens.
+     */
+    static void allow(Expression exp)
+    {
+        if (exp && exp.op == TOK.TOKcomma)
```

Comment:
I thought that even though it accepts `Expression` the expectation is that the expression is always of type `CommaExp` (at runtime) and it would be logical error if something else was passed as an argument.